pkg/util/pod/pod.go

```diff
@@ -104,3 +106,39 @@ func readUIntFromStringBelowBound(value string, bound int) (*int, error) {
 	}
 	return ptr.To(int(uintValue)), nil
 }
+
+func GenerateShape(podSpec corev1.PodSpec) (string, error) {
```

Comment:
```suggestion
func GenerateSpec(podSpec corev1.PodSpec) (string, error) {
```

I don't really know what you mean by "shape".

Reply:
This function was [moved](https://github.com/kubernetes-sigs/kueue/pull/3952/files#diff-dfb49586a8522fa91d733051fd3b7e4b3ff174907898cf249ab16e2620976a5dL560-L589) to the util/pod package to make it available for use in the StatefulSet controller as well.

Also renamed to `GenerateRoleHash`.